x/ibc/core/24-host/validate.go

```diff
@@ -112,25 +113,13 @@ func NewPathValidator(idValidator ValidateFn) ValidateFn {
 	}
 }
 
-// PathValidator takes in path string and validates with default identifier rules.
-// This is optimized by simply checking that all path elements are alphanumeric.
+// PathValidator takes in path string and validates with default identifier rules:
+// path consists of `/`-separated valid identifiers,
+// each identifier is between 1-64 characters and contains only alphanumeric and some allowed
+// special characters (see IsValidID).
 func PathValidator(path string) error {
```

Comment:
@fedekunze why do we have this function? 

It seems to be only used in connection genesis validating `ConnectionPaths` but from what I can tell, a connection path is simply all the connection ids attached to a client id. Instead we should be calling connection id validation. I don't think there are any `/` in a connection path as we currently use them

Reply:
I can open a followup fixing this, [here](https://github.com/cosmos/cosmos-sdk/blob/c0d723314133f96bd118e29401d99187a90ce986/x/ibc/core/03-connection/keeper/keeper.go#L169) is where we use `ConnectionPaths`